From f46e68a4bbd7673d28a699c84f6a282cdb6cc71f Mon Sep 17 00:00:00 2001
Message-ID: <f46e68a4bbd7673d28a699c84f6a282cdb6cc71f.1732837202.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1732837202.git.sharlatanus@gmail.com>
References: <cover.1732837202.git.sharlatanus@gmail.com>
From: "Artyom V. Poptsov" <poptsov.artyom@gmail.com>
Date: Sun, 24 Nov 2024 13:27:30 +0300
Subject: [PATCH 10/15] gnu: Add go-github-com-charmbracelet-bubbles.

* gnu/packages/golang-xyz.scm (go-github-com-charmbracelet-bubbles): New variable.

Change-Id: I090db91509590ca9a8cab1a209d00bcc12b524e7
Reviewed-by: Sharlatan Hellseher <sharlatanus@gmail.com>
---
 gnu/packages/golang-xyz.scm | 37 +++++++++++++++++++++++++++++++++++++
 1 file changed, 37 insertions(+)

diff --git a/gnu/packages/golang-xyz.scm b/gnu/packages/golang-xyz.scm
index 7abfe589cc..6d54c9efbd 100644
--- a/gnu/packages/golang-xyz.scm
+++ b/gnu/packages/golang-xyz.scm
@@ -1559,6 +1559,43 @@ (define-public go-github-com-cheggaaa-pb-v3
      (modify-inputs (package-propagated-inputs go-github-com-cheggaaa-pb)
        (append go-github-com-vividcortex-ewma)))))
 
+(define-public go-github-com-charmbracelet-bubbles
+  (package
+    (name "go-github-com-charmbracelet-bubbles")
+    (version "0.20.0")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/charmbracelet/bubbles")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "1qdcln01bq9lk6r33b8p5d5x850wgd8ddq57n4bg3xn76z2fd657"))))
+    (build-system go-build-system)
+    (arguments
+     (list #:import-path "github.com/charmbracelet/bubbles"))
+    (propagated-inputs
+     (list go-github-com-atotto-clipboard
+           go-github-com-charmbracelet-bubbletea
+           go-github-com-charmbracelet-harmonica
+           go-github-com-charmbracelet-lipgloss
+           go-github-com-charmbracelet-x-ansi
+           go-github-com-charmbracelet-x-exp-golden
+           go-github-com-dustin-go-humanize
+           go-github-com-lucasb-eyer-go-colorful
+           go-github-com-makenowjust-heredoc
+           go-github-com-mattn-go-runewidth
+           go-github-com-muesli-termenv
+           go-github-com-rivo-uniseg
+           go-github-com-sahilm-fuzzy))
+    (home-page "https://github.com/charmbracelet/bubbles")
+    (synopsis "TUI components for Bubble Tea library")
+    (description
+     "@code{bubbles} is a library that provide components for
+@@url{https://github.com/charmbracelet/bubbletea, Bubble Tea} applications.")
+    (license license:expat)))
+
 (define-public go-github-com-charmbracelet-lipgloss
   (package
     (name "go-github-com-charmbracelet-lipgloss")
-- 
2.46.0

